From 0f5acb0a5ee4663072e1d3302fbbe0d719c96b68 Mon Sep 17 00:00:00 2001
From: DeckerSU <support@decker.su>
Date: Sun, 19 Feb 2017 02:21:55 +0300
Subject: [PATCH 1/2] Partial Revert "Camera1 API: Support SW encoders for new
 camera HALs"

Read this:

- https://github.com/xen0n/android_frameworks_av_mtk/commit/0e8649c
- https://github.com/LineageOS/android_frameworks_av/commit/6b0795009b8f53ab771e0074b76381977d016f4b
- https://github.com/ResurrectionRemix-mtk/android_frameworks_av-rr/commit/f4f0f3d4201f573f8a522e50e4a00c1ac1c6a29d

This patch is a dirty compilation of these fixes. Main idea is to set storeMetaDataInVideoBuffers
to false everythere. But this may affect other codecs :(

Change-Id: Id7570b388a21e332eaca843a3e7d432db2453f7c
---
 include/media/stagefright/CameraSource.h            |  2 +-
 media/libmediaplayerservice/StagefrightRecorder.cpp |  2 +-
 media/libstagefright/CameraSource.cpp               | 10 ----------
 3 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/include/media/stagefright/CameraSource.h b/include/media/stagefright/CameraSource.h
index c2e75a6..f39688f 100644
--- a/include/media/stagefright/CameraSource.h
+++ b/include/media/stagefright/CameraSource.h
@@ -89,7 +89,7 @@ public:
                                           Size videoSize,
                                           int32_t frameRate,
                                           const sp<IGraphicBufferProducer>& surface,
-                                          bool storeMetaDataInVideoBuffers = true);
+                                          bool storeMetaDataInVideoBuffers = false);
 
     virtual ~CameraSource();
 
diff --git a/media/libmediaplayerservice/StagefrightRecorder.cpp b/media/libmediaplayerservice/StagefrightRecorder.cpp
index a6558ab..353bc02 100644
--- a/media/libmediaplayerservice/StagefrightRecorder.cpp
+++ b/media/libmediaplayerservice/StagefrightRecorder.cpp
@@ -1499,7 +1499,7 @@ status_t StagefrightRecorder::setupCameraSource(
         *cameraSource = AVFactory::get()->CreateCameraSourceFromCamera(
                 mCamera, mCameraProxy, mCameraId, mClientName, mClientUid, mClientPid,
                 videoSize, mFrameRate,
-                mPreviewSurface);
+                mPreviewSurface, false); // [*] Decker
     }
     AVUtils::get()->cacheCaptureBuffers(mCamera, mVideoEncoder);
     mCamera.clear();
diff --git a/media/libstagefright/CameraSource.cpp b/media/libstagefright/CameraSource.cpp
index fc45e38..b51a7b8 100644
--- a/media/libstagefright/CameraSource.cpp
+++ b/media/libstagefright/CameraSource.cpp
@@ -759,16 +759,6 @@ status_t CameraSource::startCameraRecording() {
             }
         }
 
-        err = mCamera->sendCommand(
-            CAMERA_CMD_SET_VIDEO_FORMAT, mEncoderFormat, mEncoderDataSpace);
-
-        // This could happen for CameraHAL1 clients; thus the failure is
-        // not a fatal error
-        if (err != OK) {
-            ALOGW("Failed to set video encoder format/dataspace to %d, %d due to %d",
-                    mEncoderFormat, mEncoderDataSpace, err);
-        }
-
         // Create memory heap to store buffers as VideoNativeMetadata.
         createVideoBufferMemoryHeap(sizeof(VideoNativeHandleMetadata), kDefaultVideoBufferCount);
     }
-- 
2.7.4

